pkg/pillar/cmd/zedagent/localinfo.go

```diff
@@ -137,17 +135,126 @@ func sendLocalAppInfo(ctx *getconfigContext) {
 				//throttle sending to be about once per hour
 				updateLocalAppInfoTicker(ctx, true)
 			}
-			if resp.StatusCode != http.StatusOK {
+			if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusNoContent {
 				errList = append(errList, fmt.Sprintf("SendLocal: wrong response status code: %d",
 					resp.StatusCode))
 				continue
 			}
+			if resp.StatusCode == http.StatusNoContent {
+				log.Functionf("Local server %s does not require additional app commands to execute",
+					localServerURL)
+				return nil
+			}
+			if appCmds.GetServerToken() != ctx.profileServerToken {
```

Comment:
Can we check here for empty appCmds to not have an error because of no aware about 204 return code?

Reply:
yes, it makes sense to not break local servers created before this feature... done